### PR TITLE
chore(ci): Add workflow for cypress pr integration

### PR DESCRIPTION
**Pull request by harsh-vador**
### Describe your changes:

I worked on adding a workflow for cypress PR integration that will help run a particular cypress based on PR labels.

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: .github/workflows/cypress-integration-tests-mysql-pr.yml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+concurrency:
+  group: cypress-integration-tests-mysql-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  cypress-ci-mysql:
+    if: |
+      github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+          || ( github.event_name == 'pull_request_target'
+               && contains(github.event.pull_request.labels.*.name, 'e2e')
```
Comment (chirag-madlani) [inline .github/workflows/cypress-integration-tests-mysql-pr.yml:46]: Let's remove e2e condition from here
```suggestion
               
```

File: .github/workflows/cypress-integration-tests-postgresql-pr.yml
Comment (chirag-madlani) [inline .github/workflows/cypress-integration-tests-postgresql-pr.yml:1]: Let's not add Postgres, Just MySQL will be fine

**Review by chirag-madlani — DISMISSED** [review]

File: .github/workflows/cypress-integration-tests-mysql-pr.yml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+#  limitations under the License.
+
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#apache-maven-with-a-settings-path
+
+name: MySQL PR Cypress Integration Tests
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "30 2 * * 1,3,5"
```
Comment (akash-jain-10) [inline .github/workflows/cypress-integration-tests-mysql-pr.yml:19]: Let's remove this!
```suggestion
 
```

File: .github/workflows/cypress-integration-tests-mysql-pr.yml [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+      - openmetadata-ui/**
+      - openmetadata-service/**
+      - openmetadata-clients/**
+      - common/**
+      
+permissions:
+  contents: read
+
+concurrency:
+  group: cypress-integration-tests-mysql-${{ github.head_ref || github.run_id }}
```
Comment (akash-jain-10) [inline .github/workflows/cypress-integration-tests-mysql-pr.yml:38]:
```suggestion
  group: cypress-integration-tests-pr-mysql-${{ github.head_ref || github.run_id }}
```

File: .github/workflows/cypress-integration-tests-mysql-pr.yml [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  cancel-in-progress: true
+
+jobs:
+  cypress-ci-mysql:
+    if: |
+      github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+          || ( github.event_name == 'pull_request_target'
+               && github.event.pull_request.draft == false
+             )
```
Comment (akash-jain-10) [inline .github/workflows/cypress-integration-tests-mysql-pr.yml:47]:
```suggestion
      github.event_name == 'workflow_dispatch'
          || ( github.event_name == 'pull_request_target'
               && github.event.pull_request.draft == false
             )
```

File: .github/workflows/cypress-integration-tests-mysql-pr.yml [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+      - name: Set Cypress Project Environment variable
+        id: cypress-project-id
+        run: |
+          if [ "${{ github.event_name }}" == "schedule" ]; then  
+            echo "CYPRESS_PROJECT_ID=${{ secrets.CYPRESS_SCHEDULE_PROJECT_ID }}" >> $GITHUB_OUTPUT
+            echo "CYPRESS_RECORD_KEY=${{ secrets.CYPRESS_SCHEDULE_RECORD_KEY }}" >> $GITHUB_OUTPUT
+          else  
+            echo "CYPRESS_PROJECT_ID=${{ secrets.CYPRESS_PR_PROJECT_ID }}" >> $GITHUB_OUTPUT
+            echo "CYPRESS_RECORD_KEY=${{ secrets.CYPRESS_PR_RECORD_KEY }}" >> $GITHUB_OUTPUT
+          fi
```
Comment (akash-jain-10) [inline .github/workflows/cypress-integration-tests-mysql-pr.yml:64]:
```suggestion
          echo "CYPRESS_PROJECT_ID=${{ secrets.CYPRESS_PR_PROJECT_ID }}" >> $GITHUB_OUTPUT
           echo "CYPRESS_RECORD_KEY=${{ secrets.CYPRESS_PR_RECORD_KEY }}" >> $GITHUB_OUTPUT
```

File: .github/workflows/cypress-integration-tests-mysql-pr.yml [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+      - name: Run Cypress Tests
+        uses: cypress-io/github-action@v4
+        timeout-minutes: 120
+        with:
+          install-command: yarn --frozen-lockfile --silent --ignore-scripts
+          record: true
+          parallel: true
+          working-directory: openmetadata-ui/src/main/resources/ui/
+          wait-on: "http://localhost:8585"
+          group: cypress-ci-psql-oss-${{ github.run_id }}-${{github.run_attempt}}
```
Comment (akash-jain-10) [inline .github/workflows/cypress-integration-tests-mysql-pr.yml:155]:
```suggestion
          group: cypress-ci-mysql-pr-oss-${{ github.run_id }}-${{github.run_attempt}}
```

File: .github/workflows/cypress-integration-tests-mysql-pr.yml [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+          CYPRESS_POSTGRES_PASSWORD: ${{ secrets.TEST_POSTGRES_PASSWORD }}
+          CYPRESS_POSTGRES_HOST_PORT: ${{ secrets.TEST_POSTGRES_HOST_PORT }}
+          CYPRESS_POSTGRES_DATABASE: ${{ secrets.TEST_POSTGRES_DATABASE }}
+          CYPRESS_AIRFLOW_HOST_PORT: ${{ secrets.TEST_AIRFLOW_HOST_PORT }}
+          CYPRESS_ML_MODEL_TRACKING_URI: ${{ secrets.TEST_ML_MODEL_TRACKING_URI }}
+          CYPRESS_ML_MODEL_REGISTRY_URI: ${{ secrets.TEST_ML_MODEL_REGISTRY_URI }}
+          CYPRESS_S3_STORAGE_ACCESS_KEY_ID: ${{ secrets.TEST_S3_STORAGE_ACCESS_KEY_ID }}
+          CYPRESS_S3_STORAGE_SECRET_ACCESS_KEY: ${{ secrets.TEST_S3_STORAGE_SECRET_ACCESS_KEY }}
+          CYPRESS_S3_STORAGE_END_POINT_URL: ${{ secrets.TEST_S3_STORAGE_END_POINT_URL }}
+          GREP_TAGS: "${{ steps.get_cypress_filter.outputs.FILTER }}"
```
Comment (akash-jain-10) [inline .github/workflows/cypress-integration-tests-mysql-pr.yml:205]: This should not be required anymore!

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: .github/workflows/cypress-integration-tests-mysql-pr.yml (hunk 7 above)
Comment (akash-jain-10) [inline .github/workflows/cypress-integration-tests-mysql-pr.yml:205]: This is not required anymore. Let's clean it up!

**Review by akash-jain-10 — APPROVED** [review]

**Pull request merged**